src/client.ts

```diff
@@ -3880,12 +3880,14 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
             }
             // Cache the key, if possible.
             // This is async.
-            this.cryptoBackend
-                .storeSessionBackupPrivateKey(privKey)
-                .catch((e) => {
-                    logger.warn("Error caching session backup key:", e);
-                })
-                .then(cacheCompleteCallback);
+            if (backupInfo.version != null) {
```

Comment:
I think we can move this check much earlier.  `version` is a required parameter on `GET /_matrix/client/v3/room_keys/keys/{roomId}` etc, so can throw an error at the top if it is not set.